File: codes/naeun/643_maximumaveragesubarray1.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double result = 0;

        int left = 0;
        double temp = 0;
        double bottom = (double) k;
        for(int i = 0 ; i < k; i++){
            temp += nums[i];
        }

        result = temp/bottom;
        for(int right = k; right < nums.length; right++){
            temp -= nums[left];
            left++;
            temp += nums[right];
            result = Math.max(result, temp/bottom);
        }


        return result;   
    }
}

File: codes/jiheon/643_maximumaveragesubarray1.rs
impl Solution {
    pub fn find_max_average(nums: Vec<i32>, k: i32) -> f64 {
        let k = k as usize;
        let mut max_sum: i32 = nums.iter().take(k).sum();
        let mut tmp_sum = max_sum;
        for i in k..nums.len() {
            tmp_sum = tmp_sum + nums[i] - nums[i - k];
            max_sum = max_sum.max(tmp_sum);
        }
        max_sum as f64 / k as f64
    }
}